snapcraft/internal/lxd.py

```diff
@@ -130,6 +131,18 @@ def _ensure_container(self):
         check_call([
             'lxc', 'config', 'set', self._container_name,
             'environment.LC_ALL', 'C.UTF-8'])
+        self._set_image_info_env_var()
+
+    def _set_image_info_env_var(self):
+        image_info = json.loads(check_output(
+            ['lxc', 'image', 'list', '--format=json', self._image]).decode())
+        image_info_edited = collections.OrderedDict()
+        for field in ('fingerprint', 'architecture', 'created_at'):
+            image_info_edited[field] = image_info[0][field]
+        check_call([
+            'lxc', 'config', 'set', self._container_name,
+            'environment.SNAPCRAFT_IMAGE_INFO',
```

Comment:
Maybe add a comment here like `# Set image info to be used in _annotate_snapcraft`.

Reply:
Good call. I added a comment, please check if it's clear.

Reply:
Very nice. Thank you for adding that!